src/include/com/amazonaws/kinesis/video/cproducer/Include.h

```diff
@@ -241,6 +241,20 @@ struct __ApiCallbacks {
 };
 typedef struct __ApiCallbacks* PApiCallbacks;
 
+/**
+ * Type of caching implementation
+ */
+typedef enum {
+    // No caching is specified, will make API calls
+    API_CALL_CACHE_TYPE_NONE,
+
+    // The stream should be pre-existing as DescribeStream API call will be emulated
+    API_CALL_CACHE_TYPE_ENDPOINT_ONLY,

```

Comment:
Considering we did not have endpoint and cache type all before, why do we have it now? Just trying to understand the context here.

Reply:
So, previously we had two behaviors controlled by a BOOL - either no caching at all or Endpoint ONLY caching. The third option that this adds is the caching ALL of the API calls which is necessary if the stream is not pre-created.